subete/repo.py

```diff
@@ -903,6 +917,20 @@ def code(self) -> str:
         logger.info(f"Retrieving code from {self._path}/{self._file_name}")
         return Path(self._path, self._file_name).read_text(errors="replace")
 
+    def image_type(self) -> str:
+        """
+        Determine if sample program is actual an image, and if so, what type.
+
+        Assuming you have a SampleProgram object called program, 
+        here's how you would use this method::
+
+            image_type: str = program.image_type()
+
+        :return: Image type if sample program is an image (e.g., "png"),
+            empty string otherwise
+        """
+        return imghdr.what(Path(self._path, self._file_name)) or ""
```

Comment:
Glad you found this library! Definitely makes our life easier. I noticed that it's going to be removed in 3.13, which is a ways out. But something to think about.

Reply:
According to this [source](https://tracyrenee61.medium.com/python-is-deprecating-some-old-models-in-their-3-13-version-of-the-language-3fab2ded371a), imagehdr is going to be replaced by filetype, puremagic, python-magic. Looks like these are all third-party. Curious to see if any of them get integrated in Python.